src/main/java/org/cryptomator/cryptofs/common/BackupHelper.java

```diff
@@ -64,16 +65,17 @@ private static void assertExistingBackupMatchesContent(Path backupFilePath, Byte
 		if (Files.exists(backupFilePath)) {
 			// TODO replace by Files.mismatch() when using JDK > 12
```

Comment:
While you're at it: Since we're now on JDK 17, we can make use of `Files.mismatch()`. In other words, instead of comparing file to the contents of a bytebuffer, we can refactor this method to `assertSameContent(Path backupFile, Path originalFile)` and simplify it.

Reply:
something like `if (Files.mismatch(...) != -1) { log warning }`

Reply:
Sounds good! will do